gef.py

```diff
@@ -2593,6 +2593,79 @@ def inner_f(*args, **kwargs):
     return wrapper
 
 
+def FakeExit(*args, **kwargs):
+    raise RuntimeWarning
+
+sys.exit = FakeExit
+
+def parse_arguments(required_arguments, optional_arguments):
+    """Argument parsing decorator."""
+
+    def int_wrapper(x): return int(x, 0)
+
+    def decorator(f):
+        def wrapper(*args, **kwargs):
+            parser = argparse.ArgumentParser(prog=args[0]._cmdline_, add_help=True)
+            for argname in required_arguments:
+                argvalue = required_arguments[argname]
+                argtype = type(argvalue)
+                if argtype == int:
+                    argtype = int_wrapper
+
+                if argname.startswith("-"):
+                    # optional args
+                    if argtype == bool:
+                        parser.add_argument(argname, action="store_true" if argvalue else "store_false")
+                    else:
+                        parser.add_argument(argname, type=argtype, required=True, default=argvalue)
+                else:
+                    if argtype in (list, tuple):
+                        nargs = '*'
+                        argtype = type(argvalue[0])
+                    else:
+                        nargs = '?'
+                    # positional args
+                    parser.add_argument(argname, type=argtype, default=argvalue, nargs=nargs)
+
+            for argname in optional_arguments:
+                if not argname.startswith("-"):
+                    # refuse positional arguments
+                    continue
+                argvalue = optional_arguments[argname]
+                argtype = type(argvalue)
+                if argtype == int:
```

Comment:
We should probably use `if isinstance(argvalue, int)`

Reply:
We have to collect the type anyway in `argtype` to pass as type when calling `parser.add_argument( type=argtype )`. What's wrong leaving is this way?

Reply:
Because if someone made a subtype, for example, your test would fail, but `isinstance` would pass. No very realistic in this context so it's fine.

When comparing types us `if argtype is int:`

Reply:
Fair enough, I'll update